cores/esp8266/FunctionalInterrupt.cpp

```diff
@@ -0,0 +1,24 @@
+#include <FunctionalInterrupt.h>
+
+
+// Duplicate typedefs from core_esp8266_wiring_digital_c
+typedef void (*voidFuncPtr)(void);
+
+// Helper functions for Functional interrupt routines
+extern "C" void ICACHE_RAM_ATTR __attachInterruptArg(uint8_t pin, voidFuncPtr userFunc, void*fp , int mode);
+
+// Structure for communication
+struct ArgStructure {
+	std::function<void(void)> reqFunction;
+};
+
+void interruptFunctional(void* arg)
+{
+	((ArgStructure*)arg)->reqFunction();
+}
+
+void attachInterrupt(uint8_t pin, std::function<void(void)> intRoutine, int mode)
```

Comment:
@igrr what about here? should this be std::function<void ICACHE_RAM_ATTR (void)> ?

Reply:
This is more tricky. You can't add attributes to function prototypes like that, and furthermore the invoke method of std::function (which gets called when you use operator() on the function object) is not placed into IRAM either (it's a part of libstdc++).

I think the only way to work around this issue is to implement interrupt masking for the duration of SPI flash operation (#3579), then IRAM_ATTRs will not be needed at all.